### DISCUSSION TITLE
How to use software serial (pio) with 115200bps?

### DISCUSSION THREAD

**Discussion by toreis-up** [micropython/micropython#12553] [RP2040 / Pico]
Hi, I am new to micropython.
I would like to realize software serial (UART) with PIO.
I have looked at https://github.com/micropython/micropython/blob/master/examples/rp2/pio_uart_rx.py and was able to communicate at 9600 bps. However, when I change UART_BAUD in the code and try to communicate at 115200 bps, the characters are garbled like this.
```
Hello, worldÉ=Õ	Õ¥9ØQ¯\
```
I am wondering if it is possible to make it work with PIO in micropython, since the method using hardware UART in micropython and software serial in the Arduino IDE works fine.
Thank you in advance for any guidance you may be able to provide.

（Original）
こんにちは、初めてmicropythonを触ります。
PIOでソフトウェアシリアル(UART)を実現したいと思っています。
https://github.com/micropython/micropython/blob/master/examples/rp2/pio_uart_rx.py を見て、9600bpsで通信できました。しかし、コード中のUART_BAUDを変更して115200bpsで通信しようとすると、このように文字化けしてしまいます。
```
Hello, worldÉ=Õ	Õ¥9ØQ¯\
```
micropythonでハードウェアUARTと、Arduino IDEでソフトウェアシリアルを使う方法は問題なく動作するので、micropythonでのPIOでも動作させることができないかと考えています。
ご教示のほどよろしくお願いいたします。

**Comment by FilipSLO** [comment]
Why don you check [examples](https://github.com/micropython/micropython/blob/master/examples/rp2/pio_uart_rx.py)?

**Reply by toreis-up** [reply]
I have seen that code and have successfully communicated at 9600 bps.
We are looking for a way to communicate at 115200 bps.

**Reply by FilipSLO** [reply]
Did you try to modify:
`UART_BAUD = 9600`

**Reply by toreis-up** [reply]
yes, but it shows this.
```
Hello, worldÉ=Õ	Õ¥9ØQ¯\
```
Tx is working reliably, so I think it's a Software Rx issue...
